integrations/cohere/src/haystack_integrations/components/generators/cohere/generator.py

```diff
@@ -66,4 +66,25 @@ def run(self, prompt: str):
         chat_message = ChatMessage.from_user(prompt)
         # Note we have to call super() like this because of the way components are dynamically built with the decorator
         results = super(CohereGenerator, self).run([chat_message])  # noqa
-        return {"replies": [results["replies"][0].text], "meta": [results["replies"][0].meta]}
+        return {
+            "replies": [results["replies"][0].text],
+            "meta": [results["replies"][0].meta],
+        }
+
+    @component.output_types(replies=List[str], meta=List[Dict[str, Any]])
+    async def run_async(self, prompt: str):
+        """
+        Queries the LLM asynchronously with the prompts to produce replies.
+
+        :param prompt: the prompt to be sent to the generative model.
+        :returns: A dictionary with the following keys:
+            - `replies`: A list of replies generated by the model.
+            - `meta`: Information about the request.
+        """
+        chat_message = ChatMessage.from_user(prompt)
+        # Note we have to call super() like this because of the way components are dynamically built with the decorator
+        results = await super(CohereGenerator, self).run_async([chat_message])  # noqa
+        return {
+            "replies": [results["replies"][0].text],
+            "meta": [results["replies"][0].meta],
+        }
```

Comment:
```suggestion
```

We haven't added async support to standard Generators because in the long run we plan to perhaps deprecate them in favor of Chat Generators, so we are implementing the new features in Chat Generators only.